### DISCUSSION TITLE
the customize plugin of skywalking, the span not display

### DISCUSSION THREAD

**Discussion by persistman** [apache/skywalking#11910] [Q&A]
I deploy a plugin based on the linke _https://skyapm.github.io/document-cn-translation-of-skywalking/zh/6.3.0/guides/Java-Plugin-Development-Guide.html_. but the the span  defined in plugin did not display in server trace tab. not konw what happened. pls help! @wu-sheng 

this is my code 
SkyPluginDemoInstrumentation.java

```
public class SkyPluginDemoInstrumentation extends ClassInstanceMethodsEnhancePluginDefine {
    private static final String ENHANCE_CLASS = "com.alibaba.fastjson.JSONObject";

    private static final String INTERCEPTOR_CLASS = "demo.skywalking.plugin.interceptor.SkyPluginDemoInterceptor";

    @Override
    protected ClassMatch enhanceClass() {
        return NameMatch.byName(ENHANCE_CLASS);
    }

    @Override
    public ConstructorInterceptPoint[] getConstructorsInterceptPoints() {
        return new ConstructorInterceptPoint[0];
    }

    @Override
    public InstanceMethodsInterceptPoint[] getInstanceMethodsInterceptPoints() {
        return new InstanceMethodsInterceptPoint[]{
                new InstanceMethodsInterceptPoint() {
                    @Override
                    public ElementMatcher<MethodDescription> getMethodsMatcher() {
                        return named("parseObject");
                    }

                    @Override
                    public String getMethodsInterceptor() {
                        return INTERCEPTOR_CLASS;
                    }

                    @Override
                    public boolean isOverrideArgs() {
                        return false;
                    }
                }
        };
    }
}

```

SkyPluginDemoInterceptor.java

```
public class SkyPluginDemoInterceptor implements InstanceMethodsAroundInterceptor {
    @Override
    public void beforeMethod(EnhancedInstance enhancedInstance, Method method, Object[] objects, Class<?>[] classes, MethodInterceptResult methodInterceptResult) throws Throwable {
        AbstractSpan span = ContextManager.createLocalSpan("JsonObject/parseObject");
        span.setComponent(ComponentsDefine.FASTJSON);
        span.tag(new StringTag("name"), "123456");
        span.tag(new StringTag("start"), "" + System.currentTimeMillis());
        SpanLayer.asCache(span);
    }

    @Override
    public Object afterMethod(EnhancedInstance enhancedInstance, Method method, Object[] objects, Class<?>[] classes, Object o) throws Throwable {
        AbstractSpan span = ContextManager.activeSpan();
        span.tag(new StringTag("end"), System.currentTimeMillis() + "");
        ContextManager.stopSpan();
        return o;
    }

    @Override
    public void handleMethodException(EnhancedInstance enhancedInstance, Method method, Object[] objects, Class<?>[] classes, Throwable throwable) {

    }
}

```

**Comment by wu-sheng** [comment]
FastJSON plugin is there, https://skywalking.apache.org/docs/skywalking-java/next/en/setup/service-agent/java-agent/supported-list/

**Plugin of fastjson serialization lib in optional plugin folder.**, from https://skywalking.apache.org/docs/main/latest/readme/

I am not sure why you pinged me on codes, I can't debug for you.

**Reply by persistman** [reply]
sorry pinged you directly,  I encountered a problem similar to this. I looked up the relevant documentation, but still didn't know how to solve it. just need your help. The span defined in the custom plug-in cannot be displayed in skywalking sever.

**Comment by CzyerChen** [comment]
use `StaticMethodsInterceptPoint` not `InstanceMethodsInterceptPoint`
You'd better refer the `Fastjson plugin` code 

<img width="495" alt="image" src="https://github.com/apache/skywalking/assets/22817918/db018ca5-e410-464d-8422-80160a7cf050">

**Reply by persistman** [reply]
OK, thanks。this is just a demo，I encountered a problem similar to this one. The span cannot be displayed. How to troubleshoot? No errors encountered

**Reply by CzyerChen** [reply]
You can import the plugin module in your project and run your project and debug the interceptor, but make sure the plugin loaded and activated.

**Reply by persistman** [reply]
I found the plugin is loaded and activated from the skywalking-api.log.

**Reply by CzyerChen** [reply]
If  no errors encountered in `skywalking-api.log` and start up correctly, you may have to import the plugin module in your project and debug it.

**Reply by persistman** [reply]
how to import the module to my project? the plugin jar in plugins folder of skywalking.

**Reply by persistman** [reply]
<img width="1097" alt="image" src="https://github.com/apache/skywalking/assets/22680996/459716fe-1fe7-46a3-816e-853947434f2d">
<img width="1298" alt="image" src="https://github.com/apache/skywalking/assets/22680996/5780d4d5-b7e0-4ac6-95d2-97366e394b99">

the offical fastjson interceptor:
<img width="1449" alt="image" src="https://github.com/apache/skywalking/assets/22680996/895908ea-1300-44bd-a4f6-3cdb96d5ba23">

There is not much difference between the two implementations， but But the official one can be displayed, but the customized one cannot be displayed.
<img width="1799" alt="image" src="https://github.com/apache/skywalking/assets/22680996/aa37f22f-5548-4e71-9ed4-6962df3354e4">

**Reply by persistman** [reply]
did not know why

**Reply by CzyerChen** [reply]
Your code is right, I can got the span, may be something wrong with the packaging, you have to check by yourself.

**Reply by persistman** [reply]
<img width="892" alt="image" src="https://github.com/apache/skywalking/assets/22680996/55689d01-5b4b-44ba-a39f-020e0e5256e9">

<img width="932" alt="image" src="https://github.com/apache/skywalking/assets/22680996/3ae1c3f7-5b4a-45d5-868c-2cae1049e362">
<img width="475" alt="image" src="https://github.com/apache/skywalking/assets/22680996/80ce824a-02a1-4633-b4d1-c2126a77a07b">

**Reply by persistman** [reply]
This is written based on the official pom file of skywalking plugin

**Comment by persistman** [comment]
resolved
